src/db/controller/channel.ts

```diff
@@ -0,0 +1,312 @@
+import { DB } from '..'
+import {
+  Boolean,
+  ChannelOpenCloseTable,
+  ChannelOpenCloseEvent,
+  ChannelState,
+} from 'src/types'
+import { In, WhereOptions, del, insert, select, update } from '../utils'
+import { LCDClient } from 'src/lib/lcdClient'
+import { ConnectionController } from './connection'
+import { PacketController, PacketFilter } from './packet'
+import { Database } from 'better-sqlite3'
+
+export class ChannelController {
+  static tableName = 'channel_open_close'
+  public static async feedEvents(
+    lcd: LCDClient,
+    chainId: string,
+    events: ChannelOpenCloseEvent[]
+  ): Promise<() => void> {
+    const feedFns: (() => void)[] = []
+    for (const event of events) {
```

Comment:
If the purpose of this function is handling promises in concurrently, may better to use
```suggestion
    const feedFns: Promise<(() => void)>[] = []

    for (const event of events) {
      switch (event.type) {
        case 'channel_open_init':
          feedFns.push(this.feedChannelOpenInitEvent(lcd, chainId, event))
          break
          ...
```

Reply:
The purpose of this function is that remove async function to execute all SQL in a single db transaction.

Reply:
okay if that is the case, good to use db transaction.